pandas/io/logfmt_format.py

```diff
@@ -0,0 +1,167 @@
+
+    encoding : str, optional
+        Encoding to be used by the parser
+
+    compression : str, optional
+        Compression to be used by the parser
+
+    Returns
+    -------
+    Data frame or `LogfmtReader`, if `chunksize` is specified.
+    """
+
+    compression = infer_compression(filepath_or_buffer, compression)
+    filepath_or_buffer, _, _, should_close = get_filepath_or_buffer(
+        filepath_or_buffer, encoding=encoding, compression=compression
+    )
+
+    logfmt_reader = LogfmtReader(
+        filepath_or_buffer,
+        dtype=dtype,
+        chunksize=chunksize,
+        encoding=encoding,
+        compression=compression,
+    )
+
+    if chunksize:
+        return logfmt_reader
+
+    result = logfmt_reader.read()
+    if should_close:
+        result.close()
+
+    return result
+
+
+class LogfmtReader(abc.Iterator):
+    """
+    LogfmtReader provides an interface for reading in a logfmt file.
+    """
+
+    def __init__(
+        self,
+        filepath_or_buffer: FilePathOrBuffer,
+        dtype: Optional[Dtype],
+        chunksize: Optional[int],
+        encoding: Optional[str],
+        compression: Optional[str],
+    ) -> None:
+        self.chunksize = chunksize
+        self.dtype = dtype
+        self.encoding = encoding
+        self.compression = compression
+        self.nrows_seen = 0
+        self.should_close = False
+
+        if self.chunksize is not None:
+            self.chunksize = _validate_integer("chunksize", self.chunksize, 1)
+
+        if isinstance(filepath_or_buffer, str) or self.compression is not None:
+            self.data, _ = get_handle(
+                filepath_or_buffer,
+                "r",
+                encoding=self.encoding,
+                compression=self.compression,
+            )
+            self.should_close = True
+        else:
+            self.data = filepath_or_buffer
+
+    def _get_data_from_filepath(self, path: FilePathOrBuffer):
+        return path
+
+    def read(self) -> DataFrame:
+        return concat(self)
+
+    def close(self) -> None:
+        """
+        If we opened a stream earlier, we should close it.
+
+        If an open stream or file was passed, we leave it open.
+        """
+        if self.should_close:
+            try:
+                self.data.close()
+            except (IOError, AttributeError):
+                pass
+
+    @staticmethod
+    def infer_types(lines: Iterable[Dict]) -> Generator[Dict, None, None]:
+        """Infer types for parsed logfmt lines"""
+        for line in lines:
+            for key in line:
+                try:
+                    line[key] = int(line[key])
+                except ValueError:
+                    try:
+                        line[key] = float(line[key])
+                    except ValueError:
+                        pass
+            yield line
+
+    def __next__(self):
+        import_optional_dependency("logfmt")
+        import logfmt
```

Comment:
Should this be
```suggestion
        logfmt = import_optional_dependency("logfmt")
```
?

Anyway, sorry if I'm missing some context, but is there an open issue about adding this to the API? Did you check that the core team would be open to having it?

Reply:
Changed this line, though `import logfmt` works too.

Opened an issue #34501 for enhancement request.